packages/fhir-router/src/graphql.ts

```diff
@@ -703,6 +829,79 @@ async function resolveField(source: any, args: any, _ctx: GraphQLContext, info:
   return array;
 }
 
+/**
+ * GraphQL resolver function for create requests.
+ * The field name should end with "Create" (i.e., "PatientCreate" for updating a Patient).
+ * The args should include the data to be created for the specified resource type.
+ * @param _source The source/root object. In the case of creates, this is typically not used and is thus ignored.
+ * @param args The GraphQL arguments, containing the new data for the resource.
+ * @param ctx The GraphQL context. This includes the repository where resources are stored.
+ * @param info The GraphQL resolve info. This includes the schema, field details, and other query-specific information.
+ * @returns A Promise that resolves to the created resource, or undefined if the resource could not be found or updated.
+ */
+async function resolveByCreate(
+  _source: any,
+  args: Record<string, any>,
+  ctx: GraphQLContext,
+  info: GraphQLResolveInfo
+): Promise<Resource | undefined> {
+  const fieldName = info.fieldName;
+  const resourceType = fieldName.substring(0, fieldName.length - 'Update'.length) as ResourceType;
+  const resourceArgs = args.res;
+  const resourceObject = { ...resourceArgs, resourceType };
+  const resource = await ctx.repo.createResource(resourceObject as Resource);
+  return resource;
+}
+
+// Mutation Resolvers
+
+/**
+ * GraphQL resolver function for update requests.
+ * The field name should end with "Update" (i.e., "PatientUpdate" for updating a Patient).
+ * The args should include the data to be updated for the specified resource type.
+ * @param _source The source/root object. In the case of updates, this is typically not used and is thus ignored.
+ * @param args The GraphQL arguments, containing the new data for the resource.
+ * @param ctx The GraphQL context. This includes the repository where resources are stored.
+ * @param info The GraphQL resolve info. This includes the schema, field details, and other query-specific information.
+ * @returns A Promise that resolves to the updated resource, or undefined if the resource could not be found or updated.
+ */
+async function resolveByUpdate(
+  _source: any,
+  args: Record<string, any>,
+  ctx: GraphQLContext,
+  info: GraphQLResolveInfo
+): Promise<Resource | undefined> {
+  const fieldName = info.fieldName;
+  const resourceType = fieldName.substring(0, fieldName.length - 'Update'.length) as ResourceType;
+  const resourceArgs = args.res;
+  const resourceId = args.id;
+  const updatedResource = { ...resourceArgs, resourceType, id: resourceId };
```

Comment:
Similar to above comment in `resolveByCreate`, my interpretation of the spec is that the user is supposed to pass in `resourceType` and `id` on their own.

Unlike `create`, we may want to do a check here that the IDs match.

For example, here are the checks that we do in the REST "update" operation: https://github.com/medplum/medplum/blob/main/packages/fhir-router/src/fhirrouter.ts#L83-L95

```ts
// Update resource
async function updateResource(req: FhirRequest, repo: FhirRepository): Promise<FhirResponse> {
  const { resourceType, id } = req.params;
  const resource = req.body;
  if (resource.resourceType !== resourceType) {
    return [badRequest('Incorrect resource type')];
  }
  if (resource.id !== id) {
    return [badRequest('Incorrect ID')];
  }
  const result = await repo.updateResource(resource);
  return [allOk, result];
}
```